### DISCUSSION TITLE
Wifi breaks up (ESP + Home assistant)

### DISCUSSION THREAD

**Discussion by ap20132022** [raomin/ESPAltherma#317] [Q&A]
Hi all,
since 1 month my wifi connection (ESP32) to the router breaks up (each 3 minutes). I have checked the signal quality. It's very good (-63dBm).

In the past, it keeps connect (without broken). Although at that time the signal quality was only -75dBm

Have any one an idea why it happened?

**Comment by raomin** [comment]
Do you have mesh/repeaters? Ie multiple AP broadcasting the same ssid?
In some cases, the esp32 would choose to connect to the weakest signal.
It could be worth trying to upgrade your libraries and framework and reflash your esp.

**Comment by ap20132022** [comment]
thank you very much for your help. 
I use only one router (Fritzbox). It wasn't changed lately.

I have downloaded the software and uploaded it to the esp. Let me check it and give some feedbacks in 2 days.
